# text: "Hello"
# number_int: 42 
# number_float: 21.3
# enable_boolean: true
# list_of_elements:
#     - 1
#     - 2
#     - 3
#     - 4
# dictionary: {
#     another_text: "World",
#     another_number: 12,
# }
dimensions: {
  wheel_diameter: 0.095,
  diff: 0.22,
  width: 0,
  height: 0
}


robot_state_publisher:
  type: robot_state_publisher/RobotStatePublisher
  publish_rate: 50
  source_list:
  - "joint_states/left_wheel_hinge"
  - "joint_states/right_wheel_hinge"
# mobile_base_controller:
#   type        : "diff_drive_controller/DiffDriveController"
#   left_wheel  : 'wheel_left_joint'
#   right_wheel : 'wheel_right_joint'
#   publish_rate: 50.0               # default: 50
#   pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
#   twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

#   # Wheel separation and diameter. These are both optional.
#   # diff_drive_controller will attempt to read either one or both from the
#   # URDF if not specified as a parameter
#   wheel_separation : 1.0
#   wheel_radius : 0.3

#   # Wheel separation and radius multipliers
#   wheel_separation_multiplier: 1.0 # default: 1.0
#   wheel_radius_multiplier    : 1.0 # default: 1.0

#   # Velocity commands timeout [s], default 0.5
#   cmd_vel_timeout: 2

#   # Base frame_id
#   base_frame_id: chassis #default: base_link

#   # Velocity and acceleration limits
#   # Whenever a min_* is unspecified, default to -max_*
#   linear:
#     x:
#       has_velocity_limits    : true
#       max_velocity           : 1.0  # m/s
#       min_velocity           : -0.5 # m/s
#       has_acceleration_limits: true
#       max_acceleration       : 0.8  # m/s^2
#       min_acceleration       : -0.4 # m/s^2
#       has_jerk_limits        : true
#       max_jerk               : 5.0  # m/s^3
#   angular:
#     z:
#       has_velocity_limits    : true
#       max_velocity           : 1.7  # rad/s
#       has_acceleration_limits: true
#       max_acceleration       : 1.5  # rad/s^2
#       has_jerk_limits        : true
#       max_jerk               : 2.5  # rad/s^3

joy_node/dev: "/dev/input/js1"
camera_topic: "/usb_cam_node/image_raw"
cmd_vel_topic: "/cmd_vel"
ultrasonic_sensor_topic: "/ultrasonic"
radar_topic: "/radar/scan"
odom_topic: "/odom"
base_link_topic: "/chassis"
battery_topic: "/battery_state"
imu_topic: "/imu"
buffer_size: 2
encoder_ticks: 360
length_between_two_wheels: 0.22
servo: 0
euclidean_dist: 40
euclidean_color:
  - 0
  - 255
  - 0
tU: 20
tD: 95
lumU: 150
lumD: 80
# gpio_radar_list_trigger :
#   - 14
#   - 18
# gpio_radar_list_echo :
#   - 4
#   - 17
# r_u_positions: 
#   - "left"
#   - "right"
gpio_list_trigger :
  - 1
  - 12
  - 16
  - 20
  - 21
gpio_list_echo:
  - 5
  - 6
  - 13
  - 19
  - 26
positions:
  - "left"
  - "front_left"
  - "front"
  - "front_right"
  - "right"
rate: 15
# 1.8152138873693582 fps aprox
joint_frames:
  - "left_wheel"
  - "right_wheel"
